test/unit/specs/components/Identity/createComponent.spec.js

```diff
@@ -103,9 +112,33 @@ describe("Identity::createComponent", () => {
     expect(customerIds.sync).toHaveBeenCalledWith(ids);
   });
 
-  it("getEcid command should make a request if no ecid value is available", () => {
+  it("getEcid command should make a request when ecid is not available", () => {
+    component.commands.getEcid.run();
```

Comment:
In this test and the next one you need to verify that the promise returned from run resolves with the ecid.  This would catch the missing return that Joe pointed out.